File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
    let lp = 0;
    let rp = arr.length-1;
    while(lp < rp){
       let mid = lp+Math.floor((rp-lp)/2);
        if(arr[mid] < arr[mid+1]){
            lp = mid+1;
        }else{
            rp=mid;
        }
    }
    return rp;
};

File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int si = 0;
        int ei = arr.length-1;
        while(si <= ei){
            int mid = si+(ei-si)/2;
            if(arr[mid] >= arr[mid+1] && arr[mid] >= arr[mid-1]){
                return mid;
            }else if( arr[mid] < arr[mid+1]){
                si = mid+1;
            }else {
                ei = mid-1;
            }
        }
        return -1;
    }
}